lib/CL/devices/devices.c

```diff
@@ -431,9 +431,11 @@ pocl_init_devices()
             pocl_devices[dev_index].available = 0;
           }
 
-          if (dev_index == 0)
-            pocl_devices[dev_index].type |= CL_DEVICE_TYPE_DEFAULT;
-
+          if (pocl_get_bool_option ("POCL_WORKAROUND_DEVTYPE", 0) == 0)
```

Comment:
Yes, I know. But your new name is not still descriptive. "What bug?" people ask. You basically add reasons for the env / history to its name which is not interesting in longer term :)

POCL_NO_DEFAULT_DEVICE_TYPE_FLAG  (that explicitly tells what it does) or similar or the one I proposed, but with the default of 1. I'm not sure if it's mandatory according to the specs to use/return the default device flag actually. Might be not as I've seen several codes with this "bug".